internal/connectors/db/yql/queries/read.go

```diff
@@ -0,0 +1,122 @@
+package queries
+
+import (
+	"errors"
+	"fmt"
+	"strings"
+)
+
+var (
+	AllBackupFields = []string{
+		"id", "container_id", "database",
+		"initiated", "created_at", "compileted_at",
+		"s3_bucket", "s3_key_prefix", "status",
+		"paths", "operation_id",
+	}
+)
+
+type StringLike interface {
+	~string
+}
+
+type QueryFilter[T StringLike] struct {
+	Field  string
+	Values []T
+	IsLike bool
+}
+
+type ReadTableQuery interface {
+	MakeFilterString() string
+	FormatQuery() (*string, error)
+}
+
+type ReadTableQueryImpl struct {
+	tableName    string
+	selectFields []string
+	filters      [][]string
+	filterFields []string
+	isLikeFilter map[string]bool
+}
+
+type ReadTableQueryOption func(*ReadTableQueryImpl)
+
+func MakeReadTableQuery(options ...ReadTableQueryOption) *ReadTableQueryImpl {
+	d := &ReadTableQueryImpl{}
+	d.filters = make([][]string, 0)
+	d.filterFields = make([]string, 0)
+	d.isLikeFilter = make(map[string]bool)
+	for _, opt := range options {
+		opt(d)
+	}
+	return d
+}
+
+func WithTableName(tableName string) ReadTableQueryOption {
+	return func(d *ReadTableQueryImpl) {
+		d.tableName = tableName
+	}
+}
+
+func WithSelectField(field string) ReadTableQueryOption {
+	return func(d *ReadTableQueryImpl) {
+		d.selectFields = append(d.selectFields, field)
+	}
+}
+
+func WithSelectFields(fields ...string) ReadTableQueryOption {
+	return func(d *ReadTableQueryImpl) {
+		d.selectFields = fields
+	}
+}
+
+func WithQueryFilters[T StringLike](filters ...QueryFilter[T]) ReadTableQueryOption {
+	return func(d *ReadTableQueryImpl) {
+		for _, filter := range filters {
+			d.filterFields = append(d.filterFields, filter.Field)
+			newFilters := make([]string, 0, len(filter.Values))
+			for _, value := range filter.Values {
+				newFilters = append(newFilters, string(value))
+			}
+			d.filters = append(d.filters, newFilters)
+			if filter.IsLike {
+				d.isLikeFilter[filter.Field] = true
+			}
+		}
+	}
+}
+
+func (d *ReadTableQueryImpl) MakeFilterString() string {
+	filterStrings := make([]string, 0, len(d.filters))
+	for i := 0; i < len(d.filterFields); i++ {
+		fieldFilterStrings := make([]string, 0, len(d.filters[i]))
+		op := "="
+		if d.isLikeFilter[d.filterFields[i]] {
+			op = "LIKE"
+		}
+		for _, value := range d.filters[i] {
+			fieldFilterStrings = append(fieldFilterStrings, fmt.Sprintf("%s %s '%s'", d.filterFields[i], op, value))
```

Comment:
should we use some escape for value ?

Reply:
quotes are used now. Maybe we could pass query parameters for values, but it seems that all them would be cached anyway, and the code will be more complex. If anyone has a strong opinion, I could introduce query params for selects, but it does not seem necessary.

Reply:
hmm, whats happens if client sends a request with filter containing single-quote (') ?

Reply:
client does not set the filters hisself, we do it when we query ydbcp db

Reply:
I've introduced table.ValueParams for `value`